File: backend/prisma/scripts/export_sqlite.js
const { PrismaClient } = require('@prisma/client');
const fs = require('fs');

async function main() {
  // Create a Prisma client pointing specifically to the SQLite file
  const prisma = new PrismaClient({
    datasources: {
      db: {
        url: 'file:../dev.db', // Path from scripts folder to prisma folder
      },
    },
  });

  const exportData = {};
  // Models to export (Mapped to SQLite table names)
  const models = {
    voter: 'Voter',
    documentType: 'DocumentType',
    voterDocument: 'VoterDocument',
    token: 'Token',
    officer: 'Officer',
    booth: 'Booth',
    auditLog: 'AuditLog',
    fingerprintTemplate: 'FingerprintTemplate'
  };

  console.log('Starting data export from SQLite...');

  for (const [prismaModel, sqliteTable] of Object.entries(models)) {
    try {
      // We use raw query bypass mapping issues
      exportData[prismaModel] = await prisma.$queryRawUnsafe(`SELECT * FROM "${sqliteTable}"`);
      console.log(`- Exported ${exportData[prismaModel].length} rows from ${sqliteTable}`);
    } catch (err) {
      console.warn(`! Skipping ${sqliteTable}: ${err.message}`);
    }
  }

  fs.writeFileSync('./prisma/backup_sqlite.json', JSON.stringify(exportData, null, 2));
  console.log('\nSUCCESS: Backup saved to ./prisma/backup_sqlite.json');
  await prisma.$disconnect();
}

main().catch(console.error);

File: backend/prisma/scripts/export_sqlite.py
import sqlite3
import json
import os

db_path = 'prisma/dev.db'
backup_path = 'prisma/backup_sqlite.json'

if not os.path.exists(db_path):
    print(f"Error: {db_path} not found.")
    exit(1)

conn = sqlite3.connect(db_path)
conn.row_factory = sqlite3.Row  # Return rows as dictionaries
cursor = conn.cursor()

# Map Prisma model names to SQLite table names
models = {
    'voter': 'Voter',
    'documentType': 'DocumentType',
    'voterDocument': 'VoterDocument',
    'token': 'Token',
    'officer': 'Officer',
    'booth': 'Booth',
    'auditLog': 'AuditLog',
    'fingerprintTemplate': 'FingerprintTemplate'
}

export_data = {}

print("Starting data export from SQLite using Python...")

for prisma_model, table_name in models.items():
    try:
        cursor.execute(f'SELECT * FROM "{table_name}"')
        rows = [dict(row) for row in cursor.fetchall()]
        export_data[prisma_model] = rows
        print(f"- Exported {len(rows)} rows from {table_name}")
    except sqlite3.OperationalError as e:
        print(f"- Skipping {table_name}: {e}")

with open(backup_path, 'w', encoding='utf-8') as f:
    json.dump(export_data, f, indent=2, default=str)

print(f"\nSUCCESS: Backup saved to {backup_path}")
conn.close()
